### PR TITLE
Bump versions and add CircleCI orbs

### PR DESCRIPTION
**Pull request by josemiguelmelo**

### CODE REVIEW

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 1]
```diff
@@ -96,8 +88,7 @@ object Tailor extends Tool {
     }.flatten
   }
 
-  private def parseToolResult(path: Path,
-                              output: List[String]): Try[List[Result]] = {
+  private def parseToolResult(path: Path, output: List[String]): Try[List[Result]] = {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [parseToolResult should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 2]
```diff
@@ -35,11 +30,12 @@ object Tailor extends Tool {
 
   private lazy val configFileNames = Set(".tailor.yml")
 
-  override def apply(source: Source.Directory,
-                     configuration: Option[List[Pattern.Definition]],
-                     files: Option[Set[Source.File]],
-                     options: Map[Options.Key, Options.Value])(
-      implicit specification: Specification): Try[List[Result]] = {
+  override def apply(
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [Method is longer than 50 lines](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: build.sbt [hunk 3]
```diff
@@ -4,19 +4,13 @@ name := "codacy-tailor"
 
 version := "1.0.0-SNAPSHOT"
 
-val languageVersion = "2.12.7"
+val languageVersion = "2.13.1"
```
Comment (lolgab) [inline build.sbt]: Please remove this variable

**Review by lolgab — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 3 above)
Comment (josemiguelmelo) [reply build.sbt]: Done ✅

**Review by lolgab — APPROVED** [review]

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 2 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [Method is longer than 50 lines](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 1 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [parseToolResult should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 4]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
+      }
+  }
 
-      lazy val nativeConfig = configFileNames
-        .map(name => Try(File(source.path) / name))
-        .collectFirst {
-          case Success(file) if file.isRegularFile =>
-            List("-c", file.toJava.getAbsolutePath)
-        }
+  private def getPatternsToUseFromCodacyConfig(
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [getPatternsToUseFromCodacyConfig should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 5]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
+
+      val command = commandToRun(patternsConfig, filesToLint)
+
+      runCommandOnSourceDir(command, source)
     }.flatten
   }
 
-  private def parseToolResult(path: Path,
-                              output: List[String]): Try[List[Result]] = {
+  private def parseToolResult(path: Path, output: List[String]): Try[List[Result]] = {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [parseToolResult should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 6]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
 
   private lazy val configFileNames = Set(".tailor.yml")
 
-  override def apply(source: Source.Directory,
-                     configuration: Option[List[Pattern.Definition]],
-                     files: Option[Set[Source.File]],
-                     options: Map[Options.Key, Options.Value])(
-      implicit specification: Specification): Try[List[Result]] = {
-    Try {
+  private def getNativeConfigurationFromSource(source: Source.Directory): Option[List[String]] = {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [getNativeConfigurationFromSource should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 7]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
+          }
+        List("--only=" + patternIds.mkString(",")) ++ parameters
+      case _ => List.empty[String]
+    }
+  }
 
-      val filesToLint: List[String] = files.fold(List(source.path)) { paths =>
-        paths.map(_.toString).toList
-      }
+  private def lintingConfiguration(source: Source.Directory, configuration: Option[List[Pattern.Definition]])(
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [lintingConfiguration should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 8]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
-              val msg =
-                s"""
-                   |${this.getClass.getSimpleName} exited with code ${resultFromTool.exitCode}
-                   |command: ${command.mkString(" ")}
-                   |message: ${e.getMessage}
-                   |stdout: ${resultFromTool.stdout.mkString(
-                     Properties.lineSeparator)}
-                   |stderr: ${resultFromTool.stderr.mkString(
-                     Properties.lineSeparator)}
+  private def failureMessage(command: List[String], resultFromTool: CommandResult, e: Throwable): String = {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [failureMessage should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 9]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
 
   private lazy val configFileNames = Set(".tailor.yml")
 
-  override def apply(source: Source.Directory,
-                     configuration: Option[List[Pattern.Definition]],
-                     files: Option[Set[Source.File]],
-                     options: Map[Options.Key, Options.Value])(
-      implicit specification: Specification): Try[List[Result]] = {
-    Try {
+  private def nativeConfigurationFromSource(source: Source.Directory): Option[List[String]] = {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [nativeConfigurationFromSource should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 10]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
+
+      val command = commandToRun(patternsConfig, filesToLint)
+
+      runCommandOnSourceDir(command, source)
     }.flatten
   }
 
-  private def parseToolResult(path: Path,
-                              output: List[String]): Try[List[Result]] = {
+  private def parseToolResult(path: Path, output: List[String]): Try[List[Result]] = {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [parseToolResult should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 11]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
+      }
+  }
 
-      lazy val nativeConfig = configFileNames
-        .map(name => Try(File(source.path) / name))
-        .collectFirst {
-          case Success(file) if file.isRegularFile =>
-            List("-c", file.toJava.getAbsolutePath)
-        }
+  private def patternsToUseFromCodacyConfig(
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [patternsToUseFromCodacyConfig should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 12]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
-              val msg =
-                s"""
-                   |${this.getClass.getSimpleName} exited with code ${resultFromTool.exitCode}
-                   |command: ${command.mkString(" ")}
-                   |message: ${e.getMessage}
-                   |stdout: ${resultFromTool.stdout.mkString(
-                     Properties.lineSeparator)}
-                   |stderr: ${resultFromTool.stderr.mkString(
-                     Properties.lineSeparator)}
+  private def failureMessage(command: List[String], resultFromTool: CommandResult, e: Throwable): String = {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [failureMessage should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala [hunk 13]
```diff
@@ -35,69 +30,95 @@ object Tailor extends Tool {
...
+          }
+        List("--only=" + patternIds.mkString(",")) ++ parameters
+      case _ => List.empty[String]
+    }
+  }
 
-      val filesToLint: List[String] = files.fold(List(source.path)) { paths =>
-        paths.map(_.toString).toList
-      }
+  private def lintingConfiguration(source: Source.Directory, configuration: Option[List[Pattern.Definition]])(
```
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [lintingConfiguration should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 9 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [nativeConfigurationFromSource should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 10 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [parseToolResult should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 11 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [patternsToUseFromCodacyConfig should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 12 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [failureMessage should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 13 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [lintingConfiguration should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 9 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [nativeConfigurationFromSource should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 10 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [parseToolResult should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 11 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [patternsToUseFromCodacyConfig should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 12 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [failureMessage should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: src/main/scala/codacy/tailor/Tailor.scala (hunk 13 above)
Comment (rtfpessoa) [inline src/main/scala/codacy/tailor/Tailor.scala]: Codacy Issue found: [lintingConfiguration should have a smaller scope by using private[this]](https://app.codacy.com/gh/codacy/codacy-tailor/pullRequest?prid=4816796)

File: .circleci/config.yml [hunk 14]
```diff
@@ -1,171 +1,51 @@
...
+      - codacy/checkout_and_version:
+          write_sbt_version: true
+      - codacy/sbt:
+          name: publish_docker_local
+          cmd: |
+            sbt "set scalafmtUseIvy in ThisBuild := false;
+                 scalafmt::test;
+                 test:scalafmt::test;
+                 sbt:scalafmt::test;
+                 set name := \"$CIRCLE_PROJECT_REPONAME\";
```
Comment (lolgab) [inline .circleci/config.yml]: No needed

File: project/plugins.sbt [hunk 15]
```diff
@@ -1 +1,2 @@
-addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.0.3")
\ No newline at end of file
+resolvers += Resolver.jcenterRepo
+addSbtPlugin("com.codacy" % "codacy-sbt-plugin" % "17.1.4")
```
Comment (lolgab) [inline project/plugins.sbt]: `17.1.5` here

**Review by lolgab — APPROVED** [review]

File: .circleci/config.yml (hunk 14 above)
Comment (josemiguelmelo) [reply .circleci/config.yml]: Done ✅

File: project/plugins.sbt (hunk 15 above)
Comment (josemiguelmelo) [reply project/plugins.sbt]: Done ✅

**Pull request merged**